collect_app/src/main/java/org/odk/collect/android/formmanagement/FormMediaDownloader.kt

```diff
@@ -54,9 +66,7 @@ class FormMediaDownloader(
         return allFormVersions.map { form: Form ->
             File(form.formMediaPath, mediaFile.filename)
         }.firstOrNull { file: File ->
-            val currentFileHash = getMd5Hash(file)
-            val downloadFileHash = mediaFile.hash
-            file.exists() && currentFileHash.contentEquals(downloadFileHash)
+            file.exists()
```

Comment:
Is `allFormVersions` guaranteed to be sorted most-recent-version-first? If not, I think this will return some arbitrary version's media file. If it does, I think there should be a test that confirms it. Maybe the naming should also make that precondition explicit.

Reply:
> Is allFormVersions guaranteed to be sorted most-recent-version-first? If not, I think this will return some arbitrary version's media file. If it does, I think there should be a test that confirms it.

Good catch. Will work on that.

Reply:
Looks good to me now!

For @getodk/testers: that means if you tap to refresh on a form that has attached entities and there’s no change to the entities dataset you should no longer see “Updated At” change. If you are manually updating attachments in Aggregate you should still not see Updated At change if you upload the same file twice. However, I’d you upload a file, then replace it, then upload the first file again, you should see a changed updated at. This should be a chance from the prior behavior (which was actually kind of weird).